io/src/main/scala/sbt/io/Path.scala

```diff
@@ -88,6 +89,39 @@ final class RichFile(val asFile: File) extends AnyVal with RichNioPath {
     new LinkOptionPath(asPath, linkOption.toVector)
 }
 
+private[sbt] trait DSLExtensions extends Any {
+  def asFile: File
+  def /(g: PathFinder.**.type): Source =
```

Comment:
I think overloading `/` infix operator is too subtle.
If I am understanding this correctly, `dir / "somthing"` would return `File`, but `dir / **` would return `Source`?

Reply:
Yes, that's correct. I use the dsl heavily in this WIP commit:
https://github.com/eatkins/sbt/commit/b16600c6b1908f09daab98a3cbfc3c41496bf9aa

I'm not sure I agree that it's too subtle. For example, I think it's even more subtle that
```
watchSources ++= (sourceDirectory.value ** "*.scala").get
```
works due to the implicit conversion between files and sources.

That being said, is there an alternative syntax that you'd recommend? My main gripe is that I think that sources should be a first class citizen within sbt and right now they are a bit confusing to work. My argument is essentially that
```
watchSources += baseDirectory.value / ** / *.scala
``` 
is significantly better than
```
watchSources += Source(baseDirectory.value, "*.scala", NothingFilter, recursive = true)
```
One alternative that I can think of that I don't love but is maybe less obtuse is
```
watchSources += baseDirectory.value.recursive.include(*.scala)
```

Reply:
I should add that I like the new syntax because it matches how I would describe the sources using globbing in bash or zsh. I think this makes the conceptual overhead of reading the build definitions lower even if the implementation is slightly harder to understand. FWIW, go to definition works (at least in intellij) so it's pretty easy to figure out what's going on.

Reply:
> That being said, is there an alternative syntax that you'd recommend?

I'd suggest something like this:

```scala
watchSources += baseDirectory.value watchRecursive withExtension("*.scala")
```

Reply:
Maybe we can compromise and provide both? I'd even be willing to put the `*` class that uses Dynamics behind an import, e.g. `import sbt.source.syntax._`.